src/shared/options.ts

```diff
@@ -119,7 +119,14 @@ export class Options {
             'omnisharp.useEditorFormattingSettings',
             true
         );
-        const enableRoslynAnalyzers = Options.readOption<boolean>(config, 'omnisharp.enableRoslynAnalyzers', false);
+        const diagnosticAnalysisScope = Options.readOption<string>(
+            config,
+            'dotnet.backgroundAnalysis.analyzerDiagnosticsScope',
+            'openFiles'
+        );
+        const enableRoslynAnalyzers =
```

Comment:
Instead of having a separate value for `diagnosticAnalysisScope` and the O# ones (`enableRoslynAnalyzers`, `analyzeOpenDocumentsOnly`)  - can we convert the O# ones into a `diagnosticAnalysisScope` value?

So if there is no value set for any of the options we respect the default value of `diagnosticAnalysisScope`
If there is a value set for `diagnosticAnalysisScope` we use that.
If there is no value for `diagnosticAnalysisScope` but there are values for O# options, we convert them to `diagnosticAnalysisScope` and set that option value.

And then if we add some code on migrateOptions, we could even remove the O# option from the package.json, leaving us with 1 unified option.